homeassistant/components/lifx/light.py

```diff
@@ -231,6 +231,9 @@ async def set_state(self, **kwargs: Any) -> None:
                 if power_off:
                     await self.set_power(False, duration=fade)
 
+            # Avoid state ping-pong by holding off updates as the state settles
+            await asyncio.sleep(0.3)
```

Comment:
Do we know why it's 0.3?  

We need to patch out the sleep in the tests.

Reply:
I'm not sure. I misunderstood the intent of this sleep which is why I removed it, thinking the debouncer would now handle it but it's the same sleep as that.

Reply:
I think the firmware probably has trouble accepting commands back to back. 

I was hoping there was a way we could determine when it's ready to accept another command instead of sleeping since there is always the risk we don't sleep enough or sleep more than we need to.

While sleep is always the last resort when we have no other way to determine when to do the task, it's sometimes the best we can do.

Reply:
> I think the firmware probably has trouble accepting commands back to back.

The firmware can take 20 commands per second, so I'm not sure this is the problem. It's more that Home Assistant and the firmware disagree on what the current state is, or that the bulb is designed to move from state to state faster than Home Assistant assumes.

Reply:
It could also be that `aiolifx` is using the response as state, which we know will be wrong for state changes that change the visible appearance of the bulb. This is a theory. I'd need to double-check the `aiolifx` code more carefully to verify.

Reply:
I originally implemented this value without fully knowing why, I thought the bulb was just slow in storing a new state. But I have since realized that the default fade in/out duration (in the bulb) is about 250ms. Hence, I now think that for the first 250ms after sending an on/off command we will get the old state – and we risk interference with the default transition, as my issue shows.

This used to cause the UI button to flap after toggling it but I think that this particular glitch has now been addressed in the UI.

Reply:
It's more subtle than that. From a former LIFX developer:

The general rule with `State` packets is if the `Set` changes the visual appearance of the device, then you’ll likely get back the state _before_ the change, otherwise you’ll likely get the new state. For this reason it’s not useful to have `res_required=True` and it’s likely you just want `ack_required=True`. 

Ref: https://photons.delfick.com/interacting/sender_interface.html#receiving-packets

Reply:
The aiolifx library doesn't take this into account, which is why my ultimate goal is to replace it with Photons.

Reply:
I think I am talking about

> The `State` packet from a `Get` packet will always be the current state of the device.

... since that was what HA did when I implemented this, it forced an `async_udpate` after `turn_off`. But that was five years ago and everything probably has changed a lot.

Reply:
It's likely a combination of both. We've overhauled the Home Assistant bits but not the airlift stuff. So that may be another option instead of total replacement.